idaes/models/properties/general_helmholtz/components/registry.py

```diff
@@ -31,27 +25,45 @@ def __init__(
         viscosity=False,
         thermal_conductivity=False,
         surface_tension=False,
+        eos_ref=None,
+        viscosity_ref=None,
+        thermal_conductivity_ref=None,
+        surface_tension_ref=None,
     ):
         self.transport_module = transport_module
         # these are true if external functions to calculate are available
         self.viscosity = viscosity
         self.thermal_conductivity = thermal_conductivity
         self.surface_tension = surface_tension
+        self.eos_ref = (eos_ref,)
+        self.viscosity_ref = (viscosity_ref,)
+        self.thermal_conductivity_ref = (thermal_conductivity_ref,)
+        self.surface_tension_ref = (surface_tension_ref,)
 
 
 def register_helmholtz_component(
     comp_str,
     viscosity=False,
     thermal_conductivity=False,
     surface_tension=False,
+    viscosity_ref=None,
```

Comment:
Is it necessary to have these as arguments to the method, or could we deduce these from which files/functions are defined? I can see arguments either way regarding the chance to miss things if an argument is set incorrectly or a file name is misspelled.

Reply:
I'm not sure.  In this there is a new auto register function that does what you suggest, and I think the need to register components one-by-one is gone.